lib/init-steps/s2-choose-strategy.js

```diff
@@ -0,0 +1,258 @@
+    }else{ //if no mobile project id, will import.
+        console.log('A valid backend specification is detected in this project')
+        result = chooseImportOrCreate(initInfo)
+    }
+
+    return result
+}
+
+function chooseForProjectInfoValid(initInfo){
+    let result = initInfo
+    initInfo.strategy = undefined
+
+    console.log('this project\'s backend is currently set to be ' + chalk.blue(initInfo.projectInfo.BackendProjectName))
+    console.log('with mobile project id = ' + chalk.blue(initInfo.projectInfo.BackendProjectID))
+    console.log('and was initialized at ' + chalk.blue(initInfo.projectInfo.InitializationTime))
+
+    if(initInfo.mobileProjectID){
+        if(initInfo.mobileProjectID != initInfo.projectInfo.BackendProjectID){
+            result = confirmToSwitchBackend(initInfo).then((initInfo)=>{
+                if(!initInfo.strategy){ //user declined to switch backend
+                    return confirmToReEstablishBackend(initInfo)
+                }else{
+                    return initInfo
+                }
+            })
+        }else{
+            prepReEstablish(initInfo)
+        }
+    }else{
+        result = chooseReEstablishOrCreate(initInfo)
+    }
+
+    return result
+}
+
+function chooseForValid(initInfo){
+    let result = initInfo
+    initInfo.strategy = undefined
+
+    console.log('this project\'s current backend is ' + chalk.blue(initInfo.projectInfo.BackendProjectName))
+    console.log('with mobile project id = ' + chalk.blue(initInfo.projectInfo.BackendProjectID))
+    console.log('and was initialized at ' + chalk.blue(initInfo.projectInfo.InitializationTime))
+
+    if(initInfo.mobileProjectID){
+        if(initInfo.mobileProjectID != initInfo.projectInfo.BackendProjectID){
+            result = confirmToSwitchBackend(initInfo)
+        }else{
+            console.log('you have specified the same id: ' + chalk.blue(initInfo.projectInfo.BackendProjectID))
+            console.log('init is aborted')
+            console.log(chalk.gray('# to retrieve the latest details of the backend awsmobile project'))
+            console.log('    $ awsmobile pull')
+        }
+    }else{
+        result = confirmToCreateNewBackend(initInfo)
+    }
+
+    return result
+}
+
+function confirmToCreateNewBackend(initInfo){
+    initInfo.strategy = undefined
+
+    let question = {
+        type: 'confirm',
+        name: 'confirmCreateNew',
+        message: 'create a new awsmobile project as the backend',
+        default: false
+    }
+
+    return inquirer.prompt(question).then(function (answers) {
+        if(answers.confirmCreateNew){
+            prepCreateNew(initInfo)
+        }
+        return initInfo
+    })
+}
+
+function confirmToReEstablishBackend(initInfo){
+    initInfo.strategy = undefined
+
+    let question = {
+        type: 'confirm',
+        name: 'confirmReEstablish',
+        message: 're-establish association with the original backend awsmobile project',
```

Comment:
Is a little confusing, maybe say something like: Use or connect to existing awsmobile project resources